pkg/internal/token/ropc.go

```diff
@@ -41,11 +55,57 @@ func newResourceOwnerToken(oAuthConfig adal.OAuthConfig, clientID, username, pas
 		resourceID:  resourceID,
 		tenantID:    tenantID,
 		oAuthConfig: oAuthConfig,
+		popClaims:   popClaims,
 	}, nil
 }
 
+// Token fetches an azcore.AccessToken from the Azure SDK and converts it to an adal.Token for use with kubelogin.
 func (p *resourceOwnerToken) Token(ctx context.Context) (adal.Token, error) {
+	return p.TokenWithOptions(ctx, nil)
+}
+
+func (p *resourceOwnerToken) TokenWithOptions(ctx context.Context, options *azcore.ClientOptions) (adal.Token, error) {
```

Comment:
`TokenWithOptions()` is not part of the interface. Can you make it private? I will refactor the rest to match this.

Reply:
Updated to make this private to the package